### PR TITLE
Add prometheus endpoint to apiserver-proxy

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Implement prometheus monitoring collector in apiserver-proxy
- Add serviceMonitor to deployment
- Add new dashboard

**Related issue(s)**

### CODE REVIEW

File: resources/monitoring/charts/grafana/dashboards/apiserver-proxy.json [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+      "percentage": false,
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "sort_desc(sum by (pod_name) (rate(container_network_receive_bytes_total{pod_name=~\"apiserver-proxy-metrics.*\"}[1m])))",
```
Comment (jakkab) [inline resources/monitoring/charts/grafana/dashboards/apiserver-proxy.json]: is `apiserver-proxy-metrics` a correct pod name?

Comment (Demonsthere) [reply resources/monitoring/charts/grafana/dashboards/apiserver-proxy.json]: no, probably not :)

**Pull request merged**